### ISSUE TITLE
bindgen emitted Rust code that won't compile

### ISSUE THREAD

**Issue by helloqirun** [rust-lang-nursery/rust-bindgen#951]
I am using this script (https://gist.github.com/fitzgen/187381e358f60efa8194d0b276b4d11a) for testing.
The hashtag for my bindgen version is e841f6f.

$ ./b.sh bindgen abc.h
```
clang-4.0: warning: treating 'c-header' input as 'c++-header' when in C++ mode, this behavior is deprecated [-Wdeprecated]
error[E0428]: the name `a` is defined multiple times
 --> /tmp/bindings-cwqNMA.rs:4:1
  |
3 | pub const a: ::std::os::raw::c_uint = 0;
  | ---------------------------------------- previous definition of the value `a` here
4 | pub const a: _bindgen_ty_1 = _bindgen_ty_1::a;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `a` redefined here
  |
  = note: `a` must be defined only once in the value namespace of this module

error: aborting due to previous error

Interesting: bindgen emitted Rust code that won't compile!
```

$ cat abc.h
```
enum {
  a =
#define a 0
      4
};
```

**Comment by emilio** [comment]
Nice one! This is a basically a duplicate of https://github.com/rust-lang-nursery/rust-bindgen/issues/687, so closing in favour of that one, thanks!
